mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -265,8 +275,22 @@ class ItoFCastOpConversion : public OpConversionPattern<CastOp> {
       return rewriter.notifyMatchFailure(castOp,
                                          "unsupported cast destination type");
 
-    rewriter.replaceOpWithNewOp<emitc::CastOp>(castOp, dstType,
-                                               adaptor.getOperands());
+    ValueRange opers = adaptor.getOperands();
+
+    // Convert to unsigned if it's the "ui" variant
+    // Signless is interpreted as signed, so no need to cast for "si"
+    Type actualOperandType = operandType;
+    if (isa<arith::UIToFPOp>(castOp)) {
+      actualOperandType =
+          rewriter.getIntegerType(operandType.getIntOrFloatBitWidth(),
+                                  /*isSigned=*/false);
+    }
+    Value fpCastOperand = opers.front();
```

Comment:
Does
```suggestion
    Value fpCastOperand = adaptor.getIn();
```
work? `CastOp::Adaptor` exposes the same interface as `CastOp`, and the ODS says that the input is called `in`, so `getIn()` should provide the operand.
It a bit nicer to access them by name than by position, in particular if there are multiple ones.

Reply:
Oh, I didn't notice access by name was also exposed. Otherwise yes, it works!